net/udpbroadcastrelay/main.c

```diff
@@ -0,0 +1,651 @@
+/*
+******************************************************************
+udp-broadcast-relay-redux
+    Relays UDP broadcasts to other networks, forging
+    the sender address.
+
+Copyright (c) 2017 Michael Morrison <github.com/sonicsnes>
+Copyright (c) 2003 Joachim Breitner <mail@joachim-breitner.de>
+Copyright (C) 2002 Nathan O'Sullivan
+
+This program is free software; you can redistribute it and/or
+modify it under the terms of the GNU General Public License
+as published by the Free Software Foundation; either version 2
+of the License, or (at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+******************************************************************
+*/
+
+#define MAXIFS    256
+#define MAXMULTICAST 256
+#define DPRINT  if (debug) printf
+#define IPHEADER_LEN 20
+#define UDPHEADER_LEN 8
+#define HEADER_LEN (IPHEADER_LEN + UDPHEADER_LEN)
+#define TTL_ID_OFFSET 64
+#define SIGF_TERM 0x1
+
+#include <sys/types.h>
+#include <sys/socket.h>
+#include <netinet/in_systm.h>
+#include <netinet/in.h>
+#include <netinet/ip.h>
+#include <netinet/udp.h>
+#include <errno.h>
+#include <string.h>
+#include <arpa/inet.h>
+#include <stdio.h>
+#ifdef __FreeBSD__
+#include <net/if.h>
+#include <net/if_dl.h>
+#else
+#include <linux/if.h>
+#endif
+#include <sys/ioctl.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <signal.h>
+
+
+
+static int exit_ok = 0;
+static sig_atomic_t sig_flags = 0;
+static char g_pidfile[128];
+
+/* list of addresses and interface numbers on local machine */
+struct Iface {
+    struct in_addr dstaddr;
+    struct in_addr ifaddr;
+    int ifindex;
+    int raw_socket;
+};
+static struct Iface ifs[MAXIFS];
+
+/* Where we forge our packets */
+static u_char gram[4096]=
+{
+    0x45,    0x00,    0x00,    0x26,
+    0x12,    0x34,    0x00,    0x00,
+    0xFF,    0x11,    0,    0,
+    0,    0,    0,    0,
+    0,    0,    0,    0,
+    0,    0,    0,    0,
+    0x00,    0x12,    0x00,    0x00,
+    '1','2','3','4','5','6','7','8','9','0'
+};
+
+void inet_ntoa2(struct in_addr in, char* chr, int len) {
+    char* from = inet_ntoa(in);
+    strncpy(chr, from, len);
+}
+
+void sig_term_handler(int signum, siginfo_t *info, void *ptr)
+{
+    unlink(g_pidfile);
+    exit(0);
+}
+
+void catch_sigterm()
+{
+    static struct sigaction _sigact;
+
+    memset(&_sigact, 0, sizeof(_sigact));
+    _sigact.sa_sigaction = sig_term_handler;
+    _sigact.sa_flags = SA_SIGINFO;
+
+    sigaction(SIGTERM, &_sigact, NULL);

```

Comment:
can you remove the underscore? It looks awful.

Reply:
Which one, there are several. I got this piece of code from a reference site. :)